src/helpers/copyContract.js

```diff
@@ -0,0 +1,26 @@
+export default async function copyContract(contractClass, address, txParams = {}) {
+  address = address.replace('0x', '')
+
+  // This is EVM assembly will return of the code of a foreign address.
+  //
+  // operation    | bytecode   | stack representation
+  // =================================================
+  // push20 ADDR  | 0x73 ADDR  | ADDR
+  // dup1         | 0x80       | ADDR ADDR
+  // extcodesize  | 0x3B       | ADDR 0xCS
+  // dup1         | 0x80       | ADDR 0xCS 0xCS
+  // swap2        | 0x91       | 0xCS 0xCS ADDR
+  // push1 00     | 0x60 0x00  | 0xCS 0xCS ADDR 0x00
+  // dup1         | 0x80       | 0xCS 0xCS ADDR 0x00 0x00
+  // swap2        | 0x91       | 0xCS 0xCS 0x00 0x00 ADDR
+  // extcodecopy  | 0x3C       | 0xCS
+  // push1 00     | 0x60 0x00  | 0xCS 0x00
+  // return       | 0xF3       |
+
+  const ASM_CODE_COPY = `0x73${address}803b8091600080913c6000f3`
+
+  const params = Object.assign({}, contractClass.defaults(), txParams, { to: 0x0, data: ASM_CODE_COPY })
+  const txHash = await web3.eth.sendTransaction(params)
+  const receipt = await web3.eth.getTransactionReceipt(txHash)
```

Comment:
I don't think that `getTransactionReceipt` blocks until the tx is mined, but rather it may return null if it is still pending. Can we check this manually on geth or parity? I'm also a bit worried as to how to detect these kind of errors when our entire test suite is on ganache.

Reply:
Also: keep in mind that web3 does not return promises, so the await is unneeded, since the version without callback simply blocks the entire thread (we may want to use callbacks though, or use a promisified version of web3 as buidler does).

Reply:
That's true, actually both functions accept a callback to work asynchronously. Let's do that to avoid an issue